lib/fileset_picker.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+# Copyright 2015 Google.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import encoder
+import glob
+import mpeg_settings
+import optimizer
+import os
+
+
+class Error(Exception):
+  pass
+
+
+def ChooseRates(width, framerate):
+  if width >= 1920 and framerate >= 50:
+    return [3000, 4500, 7000, 10000]
+  if width >= 1920 and framerate >= 24:
+    return [1600, 2500, 4000, 6000]
+  if width >= 832 and framerate >= 50:
+    return [512, 768, 1200, 2000]
+  if width >= 416:
+    return [384, 512, 850, 1500]
+  raise Error('Unhandled width/framerate combo: w=%d rate=%d' %
+              (width, framerate))
+
+
+def GenerateFileset(name):
+  yuvfiles = glob.glob(os.path.join(os.getenv('WORKDIR'),
```

Comment:
The YUV files directory is actually a subdirectory called "video"; WORKDIR is primarily used by shell scripts, the rest of the python code uses CODEC_WORKDIR.
I introduced the envars in an attempt to break out of the requirement that all programs always be run from the same directory (before this, they all implicitly depended on $CWD).

Not sure where to go from this; we need to be able to set the dir for testing, but not have to specify it explicitly in every binary. Having a lib module that picks up envars so that they can be injected into other places seems a bit strange too.

Perhaps note the problem in an issue and carry on?